mcs/class/System/System.Net/ServicePoint.cs

```diff
@@ -314,7 +314,7 @@ bool CheckAvailableForRecycling (out DateTime outIdleSince)
 
 				if (removeList != null) {
 					foreach (var group in removeList)
-						if (groups.ContainsKey (group.Name))
+						if (groups != null && groups.ContainsKey (group.Name))
```

Comment:
You could check groups == null outside of the loop to avoid checking it on every iteration.